### PR TITLE
feat: Separate the custom chart config into chart type and config

### LINKED ISSUE #7263: [Feature]-[344]:Custom chart config, Increase ease of use
## Summary

While using the chart widget custom config, It a difficult UX to create the final required object. To make this easy we will:
- Chart Type: Separate the  chart widget types and use all value which will accept a string and use the validation array to check the sanity of the given value
- Chart config: The config from the datasource object will be a separate field to all used to copy some configuration directly from the fusion chart examples

### PR DESCRIPTION
**Pull request by aswathkk**
## Description

Adds chart type and chart config options for custom chart

Fixes #7263 

## Type of change

- New feature (non-breaking change which adds functionality)
- Breaking change (fix or feature that would cause existing functionality to not work as expected)
- This change requires a documentation update

## How Has This Been Tested?

Manually tested by copy-pasting some chart config from [fusion chart docs](https://www.fusioncharts.com/dev/chart-attributes/candlestick).

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

## TODO
- [x] Add migration
- [ ] Change docs

## Test coverage results :test_tube:
<details><summary>:red_circle: Total coverage has decreased</summary>

    // Code coverage diff between base branch:release and head branch: fix/custom-chart-config 
Status | File | % Stmts | % Branch | % Funcs | % Lines 
 -----|-----|---------|----------|---------|------ 
 :red_circle: | total | 54.89 **(-0.01)** | 36.55 **(-0.03)** | 33.61 **(0.01)** | 55.43 **(-0.01)**
 :red_circle: | app/client/src/widgets/ChartWidget/component/index.tsx | 48.84 **(-0.57)** | 21.21 **(-0.9)** | 50 **(0)** | 46.88 **(-0.59)**
 :red_circle: | app/client/src/widgets/ChartWidget/widget/propertyConfig.ts | 52 **(-48)** | 13.33 **(-86.67)** | 87.5 **(-12.5)** | 52 **(-48)**
 :red_circle: | app/client/src/workers/validations.ts | 74.15 **(0)** | 67.36 **(-0.4)** | 64.71 **(0)** | 76.43 **(0)**</details>

### CODE REVIEW

**Comment by rahulramesha** [comment]
@aswathkk not sure the migration is working for custom Chart type, could you please verify?
image

**Comment by aswathkk** [comment]
@rahulramesha That was due to the merge conflict that came today.
It should work now as I resolved conflicts.

**Comment by aswathkk** [comment]
conclusion after the discussion:
Let the custom config field accept both types of configs so that the old application will also work.

File: app/client/src/widgets/ChartWidget/widget/propertyConfig.ts [hunk 1]
```diff
@@ -66,11 +66,30 @@ export default [
...
       },
+      {
+        helpText: "Fusion Chart Type see docs.appsmith.com",
+        placeholderText: "Fusion Chart Type",
+        propertyName: "customFusionChartType",
+        label: "Custom Chart Type",
+        controlType: "INPUT_TEXT",
+        isBindProperty: true,
+        isTriggerProperty: false,
+        validation: {
```
Comment (rahulramesha) [inline app/client/src/widgets/ChartWidget/widget/propertyConfig.ts:77]: This validation does not seem to be working. Could you make it work like, it does not validate the custom Chart type, if it is of older config and validates it if it is the newer custom config

File: app/client/cypress/integration/Smoke_TestSuite/ClientSideTests/DisplayWidgets/Custom_Chart_spec.js [hunk 2]
```diff
@@ -72,6 +72,10 @@ describe("Chart Widget Functionality around custom chart feature", function() {
     //cy.get(widgetsPage.toggleChartType).click({ force: true });
     cy.UpdateChartType("Custom Chart");
 
+    cy.get(viewWidgetsPage.customChartType)
```
Comment (rahulramesha) [inline app/client/cypress/integration/Smoke_TestSuite/ClientSideTests/DisplayWidgets/Custom_Chart_spec.js:75]: could you please cover both the scenarios

**Review by rahulramesha — CHANGES_REQUESTED** [review]

**Comment by aswathkk** [comment]
This requires maintaining a complex codebase to support existing users of the widget. It's better to do this in a separate version of the widget.
Closing this PR since we will be doing it in v2 of the widget.

**Pull request closed without merging**